lib/model/autocomplete.dart

```diff
@@ -10,12 +11,19 @@ import '../api/route/channels.dart';
 import '../generated/l10n/zulip_localizations.dart';
 import '../widgets/compose_box.dart';
 import 'algorithms.dart';
+import 'channel.dart';
 import 'compose.dart';
 import 'emoji.dart';
 import 'narrow.dart';
 import 'store.dart';
 
 extension ComposeContentAutocomplete on ComposeContentController {
+  int get _maxLookbackForAutocompleteIntent {
+    return 1 // intent character i.e., "#"
+      + 2 // some optional characters i.e., "_" for silent mention or "**"
+      + store.maxChannelNameLength;
+  }
```

Comment:
```suggestion
  int get _maxLookbackForAutocompleteIntent {
    return 1 // intent character i.e., "#"
      + 2 // some optional characters i.e., "_" for silent mention or "**"

      // Per the API doc, maxChannelNameLength is in Unicode code points.
      // We walk the string by UTF-16 code units, and there might be one or two
      // of those encoding each Unicode code point.
      + 2 * store.maxChannelNameLength;
  }
```

I think?

Reply:
Also nit: "e.g.", not "i.e.": https://www.merriam-webster.com/grammar/ie-vs-eg-abbreviation-meaning-usage-difference